plugin-server/src/worker/vm/upgrades/historical-export/export-historical-events-v2.ts

```diff
@@ -569,6 +561,66 @@ export function addHistoricalEventsExportCapabilityV2(
         }
     }
 
+    async function handleFetchError(
+        error: Error,
+        params: ExportParams,
+        payload: ExportHistoricalEventsJobPayload
+    ): Promise<void> {
+        if (error instanceof RetryError && payload.retriesPerformedSoFar + 1 < hub.HISTORICAL_EXPORTS_MAX_RETRY_COUNT) {
+            const nextRetrySeconds = retryDelaySeconds(payload.retriesPerformedSoFar)
+
+            createLog(
+                `Failed to fetch events from ${dateRange(
+                    payload.timestampCursor,
+                    payload.timestampCursor + payload.fetchTimeInterval
+                )}. Retrying in ${nextRetrySeconds}s`,
+                {
+                    type: PluginLogEntryType.Warn,
+                }
+            )
+            hub.statsd?.increment('historical_export.fetch_fail', {
+                teamId: pluginConfig.team_id.toString(),
+                plugin: pluginConfig.plugin?.name ?? '?',
+            })
+
+            await meta.jobs
+                .exportHistoricalEventsV2({
+                    ...payload,
+                    retriesPerformedSoFar: payload.retriesPerformedSoFar + 1,
+                } as ExportHistoricalEventsJobPayload)
+                .runIn(nextRetrySeconds, 'seconds')
```

Comment:
Do we need to spawn a new job? Couldn't we just retry on the current one?

Reply:
Saves having to add a parameter to the payload too.

Reply:
well I'd kinda want it to be in the same retry logic, because what we don't want is retry globally 14 times and within each of those retry fetch 15 times

Reply:
My point is that we are spawning an entirely new job for retrying, instead of repeating the query again in the same job. I don't see the need to spawn an entirely new job.

Reply:
> Do we need to spawn a new job? Couldn't we just retry on the current one?

We could use [graphile-worker](https://github.com/graphile/worker) retry logic. But yeah went with using the same retry as we do for the `exportEvents` failures.

Reply:
I also think it might be a rabbit hole to try and set the graphile worker retry for this.

As for retrying in the same job run, the retry backoff can be a long time, we don't want to saturate the worker pool.